tests/e2e/utils/DriverHelper.ts

```diff
@@ -194,99 +202,121 @@ export class DriverHelper {
     }
 
     public async waitAllDisappearance(locators: Array<By>,
-        attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
-        polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
+        attemptsPerLocator: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
+        pollingPerLocator: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
 
         Logger.trace(`DriverHelper.waitAllDisappearance ${locators}`);
 
         for (const elementLocator of locators) {
-            await this.waitDisappearance(elementLocator, attempts, polling);
+            await this.waitDisappearance(elementLocator, attemptsPerLocator, pollingPerLocator);
         }
     }
 
     public async waitAndClick(elementLocator: By, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.waitAndClick ${elementLocator}`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitVisibility(elementLocator, timeout);
-
             try {
-                await element.click();
-                return;
-            } catch (err) {
-                if (err instanceof error.StaleElementReferenceError) {
-                    Logger.debug(`DriverHelper.waitAndClik - ${elementLocator} - StaleElementReferenceError - ${err}`);
-                    await this.wait(polling);
-                    continue;
+                const element: WebElement = await this.waitVisibility(elementLocator, polling);
+                try {
+                    await element.click();
+                    return;
+                } catch (err) {
+                    if (err instanceof error.StaleElementReferenceError) {
+                        Logger.debug(`DriverHelper.waitAndClik - ${elementLocator} - StaleElementReferenceError - ${err}`);
+                        await this.wait(polling);
+                        continue;
+                    }
+
+                    if (err instanceof error.WebDriverError) {
```

Comment:
This one will match all times because all errors are children of the "WebDriverError"
```
/**
   * A new session could not be created.
   */
  class SessionNotCreatedError extends WebDriverError {
    /** @param {string=} opt_error the error message, if any. */
    constructor(opt_error?: string);
  }

  /**
   * An element command failed because the referenced element is no longer
   * attached to the DOM.
   */
  class StaleElementReferenceError extends WebDriverError {
    /** @param {string=} opt_error the error message, if any. */
    constructor(opt_error?: string);
  }

  /**
   * An operation did not completErrorCodee before its timeout expired.
   */
  class TimeoutError extends WebDriverError {
    /** @param {string=} opt_error the error message, if any. */
    constructor(opt_error?: string);
  }
```

Reply:
Thank you for the heads-up, but I've only copied over the original code, I can try to modify it to better match the intent, but I believe the second if statement is there to catch any other exception that may not be `StaleElementReferenceError`